pkg/kratos/handlers.go

```diff
@@ -280,19 +278,29 @@ func (a *API) handleCreateRecoveryFlow(w http.ResponseWriter, r *http.Request) {
 func (a *API) handleGetSettingsFlow(w http.ResponseWriter, r *http.Request) {
 	q := r.URL.Query()
 
-	flow, cookies, err := a.service.GetSettingsFlow(context.Background(), q.Get("id"), r.Cookies())
+	flow, cookies, redirect, err := a.service.GetSettingsFlow(context.Background(), q.Get("id"), r.Cookies())
 	if err != nil {
 		a.logger.Errorf("Error when getting settings flow: %v\n", err)
 		http.Error(w, "Failed to get settings flow", http.StatusInternalServerError)
 		return
 	}
 
-	resp, err := flow.MarshalJSON()
+	resp, err := json.Marshal(flow)
 	if err != nil {
 		a.logger.Errorf("Error when marshalling json: %v\n", err)
 		http.Error(w, "Failed to parse settings flow", http.StatusInternalServerError)
 		return
 	}
+
+	if redirect != nil {
+		resp = []byte(*redirect.RedirectTo)
```

Comment:
If it's a json it doesn't redirect:
![image](https://github.com/canonical/identity-platform-login-ui/assets/97316813/67c397f7-73e8-4b7e-b923-7cefc7a4c9e6)
I think that's because errors are handled differently in `reset_password` than in login: https://github.com/canonical/identity-platform-login-ui/pull/249/files#diff-8644072735eaac8081eaf42339bad3844a0caa4c0fe45d89a987e9d10e039284R37
In [login](https://github.com/canonical/identity-platform-login-ui/blob/3965a2f2c05aa49d0f8d446190ad07a8904fc70c/ui/pages/login.tsx#L54), the redirect url is read from a `FlowResponse` type.

I'm fine to return the json here, but can we parse the error responses in the same way @edlerd?

Reply:
Yes, we can change this format. The current implementation checks if the response body is a valid url and if that is paired with a 403 status, we redirect to that url. changing this response to a json would mean we also have to change the error handling. Let me know if you change the response body structure and I can also change the frontend error handling.

Reply:
Response body changed, it is now a json

Reply:
Changed the frontend to handle the new JSON response body instead of the plain URL in the body.